src/Service/CronEventManager.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace WP_Statistics\Service;
+
+use WP_Statistics\Components\Event;
+use WP_STATISTICS\Option;
+use WP_Statistics\Service\Admin\MarketingCampaign\MarketingCampaignFetcher;
+use WP_Statistics\Service\Admin\Notification\NotificationFetcher;
+
+class CronEventManager
+{
+    /**
+     *
+     */
+    public function __construct()
+    {
+        if (Option::get('display_notifications')) {
+            Event::schedule('wp_statistics_notification_hook', time(), 'daily', [$this, 'fetchNotification']);
+            Event::schedule('wp_statistics_marketing_campaign_hook', time(), 'daily', [$this, 'fetchMarketingCampaign']);
```

Comment:
I really not sure this is in the right statement, maybe move out the condition `if (Option::get('display_notifications')) {` is more better.